### PR TITLE
Feature/adf 304/feature flag tabular import

### PR DESCRIPTION
**Pull request by chinnu-focaloid**
feat: feature flag for tabular import

https://oat-sa.atlassian.net/browse/ADF-304

### CODE REVIEW

File: actions/class.ItemImport.php [hunk 1]
```diff
@@ -67,4 +66,30 @@ protected function getAvailableImportHandlers()
...
+                //Check feature flag for tabular import is enabled or not
+                if (!$this->getFeatureFlagChecker()->isEnabled(self::FEATURE_FLAG_TABULAR_IMPORT)) {
+                    // Remove CSV object if tabular import is disabled
+                    unset($returnValue[$key]);
+                } else {
+                    $importer = new CsvItemImporter();
+                    $importer->setServiceLocator($this->getServiceLocator());
+                    $returnValue[$key] = $importer;
+                }
+            }
```
Comment (gabrielfs7) [inline actions/class.ItemImport.php:85]: Good job! 

Just some clean code recommendations. 

- You can always avoid ELSE keyword for better readability. Code becomes smaller: https://williamdurand.fr/2013/06/03/object-calisthenics/#2-dont-use-the-else-keyword
- You can avoid comments, cause your code is self-explanatory.

```suggestion
            if ($returnValue[$key] instanceof \tao_models_classes_import_CsvImporter) {
                if ($this->getFeatureFlagChecker()->isEnabled(self::FEATURE_FLAG_TABULAR_IMPORT)) {
                    $importer = new CsvItemImporter();
                    $importer->setServiceLocator($this->getServiceLocator());
                    $returnValue[$key] = $importer;
                    
                    continue;
                }
                
                unset($returnValue[$key]);
            }
```

File: actions/class.ItemImport.php [hunk 2]
```diff
@@ -33,6 +37,8 @@
  */
 class taoItems_actions_ItemImport extends tao_actions_Import
 {
+    public const FEATURE_FLAG_TABULAR_IMPORT = 'FEATURE_FLAG_TABULAR_IMPORT';
```
Comment (gabrielfs7) [inline actions/class.ItemImport.php:40]: As we also can have feature flag do disable, maybe a suffix to be 100% clear. Just a suggestion. 

```suggestion
    public const FEATURE_FLAG_TABULAR_IMPORT = 'FEATURE_FLAG_TABULAR_IMPORT_ENABLED';
```

**Review by gabrielfs7 — APPROVED** [review]
Good job! Just left some comments as suggestions :) 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: actions/class.ItemImport.php [hunk 3]
```diff
@@ -35,6 +37,8 @@
  */
 class taoItems_actions_ItemImport extends tao_actions_Import
 {
+    public const FEATURE_FLAG_TABULAR_IMPORT = 'FEATURE_FLAG_TABULAR_IMPORT_ENABLED';
```
Comment (boajer) [inline actions/class.ItemImport.php:40]: Why should it be public?

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by voknil — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified on http://tabular-adf-156-items-import.playground.kitchen.it.taocloud.org:46001/

**Pull request merged**